crates/flipperzero/src/subghz/device.rs

```diff
@@ -0,0 +1,193 @@
+//! Safe wrapper utilizing the API provided in `lib/subghz/devices/devices.h`
+
+use super::{error::SubGhzError, preset::SubGhzPreset};
+use core::{ffi::CStr, ptr::null_mut};
+use flipperzero_sys::{
+    subghz_devices_begin, subghz_devices_deinit, subghz_devices_end, subghz_devices_flush_rx,
+    subghz_devices_flush_tx, subghz_devices_get_by_name, subghz_devices_idle, subghz_devices_init,
+    subghz_devices_is_connect, subghz_devices_is_frequency_valid,
+    subghz_devices_is_rx_data_crc_valid, subghz_devices_load_preset, subghz_devices_read_packet,
+    subghz_devices_reset, subghz_devices_rx_pipe_not_empty, subghz_devices_set_frequency,
+    subghz_devices_set_rx, subghz_devices_set_tx, subghz_devices_sleep,
+    subghz_devices_write_packet, SubGhzDevice,
+};
+use uom::si::{frequency::hertz, u32::Frequency};
+
+pub struct SubGhz {
+    device: &'static SubGhzDevice,
```

Comment:
Storing `*const SubGhzDevice` as a Rust reference is really tricky, considering that there are much more invariant to uphold for it, especially given that `SubGhzDevice` seems to have some form of interior mutability internally. I suggest that you store `NonNull<SubGhzDevice>` instead, only creating short-living references when needed (e.g. when reading fields).

Reply:
So looking at [this instance of a SubGhzDevice](https://github.com/flipperdevices/flipperzero-firmware/blob/fe424061c968c41131448697ea587533402dccae/lib/subghz/devices/cc1101_int/cc1101_int_interconnect.c#L93), it looks like a given SubGhzDevice is going to be a struct that contains a name and a pointer to a struct that contains a bunch of function pointers, so I don't think this is an issue; but I also do not have much practical experience writing in C, so I could not be interpreting things correctly. All that being said, making the change back to `*const SubGhzDevice` is simple and only effect the `get_rssi` ang `get_lqi` functions, though I am not sure how should now change the semantics of those functions: ie should they still be `Option<T>` or `Result<T>` to indicate if the device pointer is null or if the function pointer is null.